src/views/studio/StudioLayout.tsx

```diff
@@ -10,61 +10,171 @@ import {
   { path: relativeRoutes.studio.uploads(), element: <MyUploadsView /> },
 ]
 
 const StudioLayout = () => {
   const navigate = useNavigate()
+  const location = useLocation()
   const { isUserConnectedToInternet, nodeConnectionStatus } = useConnectionStatus()
+  const { extensionConnected, extensionConnectionLoading } = useJoystream()
+
+  const {
+    activeUser: { accountId, memberId, channelId },
+    setActiveChannel,
+    loading: activeUserLoading,
+  } = useActiveUser()
+  const { membership, loading: membershipLoading, error: membershipError } = useMembership(
+    {
+      where: { id: memberId },
+    },
+    {
+      skip: !memberId,
+    }
+  )
+
+  const [enterLocation] = useState(location.pathname)
+
+  const authenticated = !!accountId && !!memberId && !!channelId && extensionConnected
+
+  useEffect(() => {
+    if (extensionConnectionLoading) {
+      return
+    }
+    if (!extensionConnected) {
+      navigate(absoluteRoutes.studio.join())
+    }
+  }, [extensionConnected, extensionConnectionLoading, navigate])
+
+  useEffect(() => {
+    if (activeUserLoading || channelId || !extensionConnected) {
+      return
+    }
+    if (!accountId) {
+      navigate(absoluteRoutes.studio.join({ step: '2' }))
+      return
+    }
+    if (!memberId) {
+      navigate(absoluteRoutes.studio.signIn())
+    }
+  }, [accountId, activeUserLoading, channelId, extensionConnected, memberId, navigate])
+
+  useEffect(() => {
+    if (activeUserLoading || channelId || !extensionConnected || !memberId || !accountId) {
+      return
+    }
+
+    // TODO add lastChannelId and setting that to activeChannel
+
+    if (membershipLoading) {
+      return
+    }
+    if (!membership?.channels.length) {
+      navigate(absoluteRoutes.studio.newChannel())
+      return
+    }
+    setActiveChannel(membership.channels[0].id)
+    navigate(enterLocation)
+  }, [
+    accountId,
+    activeUserLoading,
+    channelId,
+    enterLocation,
+    extensionConnected,
+    memberId,
+    membership,
+    membershipLoading,
+    navigate,
+    setActiveChannel,
+  ])
+
+  useEffect(() => {
+    if (activeUserLoading || !authenticated) {
+      return
+    }
+    if (location.pathname === '/studio/') {
+      navigate(absoluteRoutes.studio.videos())
+    }
+  }, [activeUserLoading, authenticated, location, navigate])
+
+  if (membershipError) {
+    throw membershipError
+  }
 
   // TODO: add route transition
   // TODO: remove dependency on PersonalDataProvider
   //  we need PersonalDataProvider because DismissibleMessage in video drafts depends on it
 
+  return (
+    <>
+      <NoConnectionIndicator
+        nodeConnectionStatus={nodeConnectionStatus}
+        isConnectedToInternet={isUserConnectedToInternet}
+      />
+      <StudioTopbar hideChannelInfo={!authenticated} />
+      {authenticated && <StudioSidenav />}
+      {extensionConnectionLoading ? (
+        <LoadingStudio />
```

Comment:
This is only displayed when `extensionConnectionLoading`, but I think per flow it should also be displayed if `accountId && memberId` are set but `channelId` is not and we're still fetching the member right? This is `wait till member details are fetched` in the diagram. Or we don't need it for some reason?